mirrord/protocol/src/outgoing.rs

```diff
@@ -1,17 +1,130 @@
 use core::fmt;
-use std::net::SocketAddr;
+use std::{
+    fmt::{Display, Formatter},
+    io,
+    io::ErrorKind,
+    net::SocketAddr,
+    path::PathBuf,
+};
 
 use bincode::{Decode, Encode};
+use socket2::SockAddr;
 
-use crate::ConnectionId;
+use crate::{
+    outgoing::UnixAddr::{Abstract, Pathname},
+    ConnectionId, SerializationError,
+};
 
 pub mod tcp;
 pub mod udp;
 
+/// A serializable socket address type that can represent IP addresses or addresses of unix sockets.
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone)]
+pub enum SocketAddress {
```

Comment:
`SocketAddr` is IP only (thought there is only the unix-only [`SocketAddr`](https://github.com/metalbear-co/mirrord/pull/1167#discussion_r1134275107)).
`SockAddr` is all C/OSy - it uses an array of bytes on which it stores the C socket address type.
`SocketAddress` is our type that is both IP and unix, and is also made with normal Rust types, and is serializable for our protocol. Should we give it a different name? Do you have a suggestion?

Reply:
`MirrordSocketAddress`?
`IpOrUnixSocketAddress`?

Reply:
Added aliases in https://github.com/metalbear-co/mirrord/pull/1167/commits/f4ac948666d42b2df3fe221c99c5c049b2778108.
What do you mean by 
> Is it possible to use them as e.g. socket2::SockAddr in the code?

Reply:
I meant actually spell out socket2::SockAddr, rather than import it at the top and then just use SockAddr. This would make it clear you're just converting between types in different libraries that are meant to represent the same thing. But aliases work too